server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java

```diff
@@ -571,6 +573,31 @@ public void test_add_column_fixes_inner_types_of_all_its_parents() throws Except
             .hasType(oType);
     }
 
+    @Test
+    public void test_add_columns_fails_eagerly_on_too_many_columns() throws Exception {
+        SQLExecutor e = SQLExecutor.of(clusterService)
+            .addTable("create table tbl (x int) with (\"mapping.total_fields.limit\" = 3)");
+        DocTableInfo table = e.resolveTableInfo("tbl");
+        Function<String, Reference> newRef = name -> new SimpleReference(
+            new ReferenceIdent(table.ident(), ColumnIdent.of(name)),
+            RowGranularity.DOC,
+            DataTypes.INTEGER,
+            -1,
+            null
+        );
+        Reference a = newRef.apply("a");
+        Reference b = newRef.apply("b");
+        Reference c = newRef.apply("c");
+        assertThatThrownBy(() -> table.addColumns(
```

Comment:
Shouldn't we add `a` and `b` in one step, and then the `c`, so that we also test that the syscolumns are not counted?

Reply:
There was another test failing because I had initially left out the sys-column extraction, so that's covered already.

Reply:
cool, thx.